tiledb/sm/group/group.cc

```diff
@@ -431,68 +414,66 @@ Status Group::get_metadata(
     const void** value) {
   // Check if group is open
   if (!is_open_)
-    return Status_GroupError("Cannot get metadata; Group is not open");
+    throw GroupStatusException("Cannot get metadata; Group is not open");
 
   // Check mode
   if (query_type_ != QueryType::READ)
-    return Status_GroupError(
+    throw GroupStatusException(
         "Cannot get metadata; Group was "
         "not opened in read mode");
 
   // Load group metadata, if not loaded yet
   if (!metadata_loaded_)
-    RETURN_NOT_OK(load_metadata());
+    load_metadata();
 
-  RETURN_NOT_OK(
+  throw_if_not_ok(
       metadata_.get(index, key, key_len, value_type, value_num, value));
-
-  return Status::Ok();
 }
 
-Status Group::get_metadata_num(uint64_t* num) {
+uint64_t Group::metadata_num() {
   // Check if group is open
   if (!is_open_)
-    return Status_GroupError(
+    throw GroupStatusException(
         "Cannot get number of metadata; Group is not open");
 
   // Check mode
   if (query_type_ != QueryType::READ)
-    return Status_GroupError(
+    throw GroupStatusException(
         "Cannot get number of metadata; Group was "
         "not opened in read mode");
 
   // Load group metadata, if not loaded yet
   if (!metadata_loaded_)
-    RETURN_NOT_OK(load_metadata());
+    load_metadata();
 
-  *num = metadata_.num();
-
-  return Status::Ok();
+  return metadata_.num();
 }
 
-Status Group::has_metadata_key(
-    const char* key, Datatype* value_type, bool* has_key) {
+std::optional<Datatype> Group::get_metadata_type_if_exists(const char* key) {
```

Comment:
`get` in the name is not needed in the name at all.

`if_exists` in the name is redundant here. The return value is `optional`. It's apparent that there's some conditionality.